sourcing-new/src/main/scala/ch/epfl/bluebrain/nexus/sourcingnew/processor/EventSourceProcessor.scala

```diff
@@ -0,0 +1,329 @@
+
+  /**
+    * Event source processor implementation relying on akka-persistence
+    * so than its state can be recovered
+    * @param entityId
+    * @param definition
+    * @param config
+    * @tparam F
+    * @tparam State
+    * @tparam EvaluateCommand
+    * @tparam Event
+    * @tparam Rejection
+    * @return
+    */
+  class PersistentEventProcessor[
+    F[_]: CatsEffect,
+    State: ClassTag,
+    EvaluateCommand: ClassTag,
+    Event: ClassTag,
+    Rejection: ClassTag](entityId: String,
+                         override val definition: PersistentEventDefinition[F, State, EvaluateCommand, Event, Rejection],
+                         override val stopStrategy: PersistentStopStrategy,
+                         stopAfterInactivity: ActorRef[ProcessorCommand] => Behavior[ProcessorCommand],
+                         config: AggregateConfig)
+        extends EventSourceProcessor[F, State, EvaluateCommand, Event, Rejection](entityId, stopAfterInactivity, config) {
+
+          import definition._
+
+          private def commandHandler(actorContext: ActorContext[EventSourceCommand]):
+          (State, EventSourceCommand) => Effect[Event, State] = {
+            (state, command) =>
+              command match {
+                case RequestState(_, replyTo: ActorRef[State]) =>
+                  Effect.reply(replyTo)(state)
+                case RequestLastSeqNr(_, replyTo: ActorRef[GetLastSeqNr]) =>
+                  Effect.reply(replyTo)(
+                    GetLastSeqNr(EventSourcedBehavior.lastSequenceNumber(actorContext))
+                  )
+                case Append(_, event: Event) => Effect.persist(event)
+              }
+          }
+
+          override protected def stateBehavior(state: State, parent: ActorRef[ProcessorCommand]): Behavior[EventSourceCommand] =  {
+            import SnapshotStrategy._
+            val persistenceId = PersistenceId.ofUniqueId(id)
+            Behaviors.setup { context =>
+              context.log.info2("Starting event source processor for type {} and id {}",
+                entityType,
+                entityId)
+              EventSourcedBehavior[EventSourceCommand, Event, State](
+                persistenceId,
+                emptyState = initialState,
+                commandHandler(context),
+                next
+              ).withTagger(tagger)
+                .snapshotStrategy(snapshotStrategy)
+                .receiveSignal {
+                case (state, RecoveryCompleted) =>
+                  // The actor will be stopped/passivated after a fix period after recovery
+                  stopStrategy.lapsedSinceRecoveryCompleted.foreach { duration =>
+                    context.scheduleOnce(duration, parent, Idle)
+                  }
+                  context.log.debugN("Entity {} has been successfully recovered at state {}",
+                    persistenceId,
+                    state)
+                case (state, RecoveryFailed(failure)) =>
```

Comment:
do we also need to apply the stop strategy here?

Reply:
This is already taken care of:
If there is a problem with recovering the state of the actor from the journal, a RecoveryFailed signal is emitted to the receiveSignal handler and the actor will be stopped (or restarted with backoff).